sharding/src/main/scala/com/lightbend/akka/bench/sharding/scalability/ActorCountingEntity.scala

```diff
@@ -0,0 +1,57 @@
+package com.lightbend.akka.bench.sharding.scalability
+
+import akka.actor.{ Actor, ActorRef, ActorSystem, Props }
+import akka.cluster.sharding.{ ClusterSharding, ClusterShardingSettings, ShardRegion }
+import com.lightbend.akka.bench.sharding.BenchSettings
+
+object ActorCountingEntity {
+
+  sealed trait HasId { def id: String }
+  // sent from master to entity
+  final case class Start(id: Integer, sentTimestamp: Long) extends HasId
+  // entity replies with that once it has started, the time is the sender's time, 
+  // so the sender can calculate how long it took for the actor to get the message
+  final case class Ready(sentTimestamp: Long)
+
+  def props() = Props[ActorCountingEntity]
+
+  // sharding config
+  val typeName = "bench-entity"
+
+  val extractEntityId: ShardRegion.ExtractEntityId = {
+    case msg: ActorCountingEntity.HasId => (msg.id, msg)
+  }
+
+  def extractShardId(numberOfEntities: Int): ShardRegion.ExtractShardId = {
+    case msg: ActorCountingEntity.HasId => (msg.id.hashCode % numberOfEntities).toString
```

Comment:
this should be `% numberOfShards`, see https://github.com/akka/akka/blob/master/akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala#L134